src/NLog/LogFactory.cs

```diff
@@ -1092,15 +1093,51 @@ public void ResetCandidateConfigFilePath()
         private static IEnumerable<string> GetDefaultCandidateConfigFilePaths()
         {
             // NLog.config from application directory
-            if (CurrentAppDomain?.BaseDirectory != null)
+            string baseDirectory = PathHelpers.TrimDirectorySeparators(CurrentAppDomain?.BaseDirectory);
+            if (!string.IsNullOrEmpty(baseDirectory))
```

Comment:
please move the check `string.IsNullOrEmpty(baseDirectory)` on top of both `if`s - it's now duplicated

(so we have 3 ifs, 1 top and 2 nested)

Reply:
Well I like to skip the `PlatformDetector.IsWin32` when possible.

Reply:
I'm a bit confused, currently it tests the `PlatformDetector.IsWin32` before the baseDirectory check, so it isn't skipped when possible?

To be clear, the proposal:


```c#

if (!string.IsNullOrEmpty(baseDirectory))
{
    yield return Path.Combine(baseDirectory, "NLog.config");
    bool platformFileSystemCaseInsensitive = PlatformDetector.IsWin32;
    if (!platformFileSystemCaseInsensitive)
    {
         yield return Path.Combine(baseDirectory, "nlog.config");
    }

}

```

`baseDirectory` was checked before, so no reason to check it again IMO

Reply:
PlatformDetector.IsWin32 is only called once in the current implementation. Your suggestions requires multiple lookups.

Reply:
OK, Fair enough